src/scripts/background.js

```diff
@@ -910,80 +916,75 @@ TogglButton = {
       }
       keys.sort();
 
-      if (TogglButton.$user.workspaces.length > 1) {
-
-        // Add Workspace names
-        TogglButton.$user.workspaces.forEach(function (element, index) {
-          wsHtml[element.id] = {};
-          wsHtml[element.id][0] = '<option disabled="disabled">  ---  ' + element.name.toUpperCase() + '  ---  </option>';
-        });
+      // Add Workspace names
+      TogglButton.$user.workspaces.forEach(function (element, index) {
+        wsHtml[element.id] = {};
+        wsHtml[element.id][0] = '<ul class="ws-list" data-wid="' + element.id + '" title="' + escapeHtml(element.name) + '"><li class="ws-row' + hideWs + '" title="' + escapeHtml(element.name.toUpperCase()) + '">' + escapeHtml(element.name.toUpperCase()) + '</li>' +
+          '<ul class="client-list" data-cid="0">';
+      });
 
-        // Add client optgroups
-        for (i = 0; i < keys.length; i++) {
-          client = clientNames[keys[i]];
-          wsHtml[client.wid][client.name + client.id] = '<optgroup label="' + client.name + '">';
-        }
+      // Add client optgroups
+      for (i = 0; i < keys.length; i++) {
+        client = clientNames[keys[i]];
+        wsHtml[client.wid][client.name + client.id] = '<ul class="client-list" data-cid="' + client.id + '"><li class="client-row" title="' + client.name + '">' + escapeHtml(client.name) + '</li>';
+      }
 
-        // Add projects
-        for (key in projects) {
-          if (projects.hasOwnProperty(key)) {
-            project = projects[key];
-            clientName = (!!project.cid && !!clients[project.cid]) ? (clients[project.cid].name + project.cid) : 0;
-            wsHtml[project.wid][clientName] += "<option value='" + project.id + "'>" + project.name + "</option>";
-          }
+      // Add projects
+      for (key in projects) {
+        if (projects.hasOwnProperty(key)) {
+          project = projects[key];
+          clientName = (!!project.cid && !!clients[project.cid]) ? (clients[project.cid].name + project.cid) : 0;
+          wsHtml[project.wid][clientName] += TogglButton.generateProjectItem(project);
         }
+      }
 
-        // create html
-        for (key in wsHtml) {
-          if (wsHtml.hasOwnProperty(key)) {
-            Object.keys(wsHtml[key]).sort();
-            for (ckey in wsHtml[key]) {
-              if (wsHtml[key].hasOwnProperty(ckey) && validate(wsHtml[key][ckey])) {
-                html += wsHtml[key][ckey] + "</optgroup>";
-              }
+      // create html
+      for (key in wsHtml) {
+        if (wsHtml.hasOwnProperty(key)) {
+          Object.keys(wsHtml[key]).sort();
```

Comment:
Does this actually sort the properties in `wsHtml[key]`? From what I can tell it shouldn't..

Apparently ES6 spec officially formalizes order of iteration on an object (http://stackoverflow.com/questions/5467129/sort-javascript-object-by-key#answer-31102605) but basically it's creation order.

Short proof Object.keys().sort() doesn't seem to work:
```
> o = {"z": 2, "a": 1}
>>> Object {z: 2, a: 1}
> Object.keys(o).sort()
>>> ["a", "z"]
> for (k in o) { console.log(k); }
>>> z
>>> a
```

=> if keys were sorted at time of creation, remove line. else, something like `var sortedKeys = Object.keys(wsHtml[key]).sort(); for (ckey in sortedKeys) { ... `